### PR TITLE
Fix a bug with excess RAM usage during vector computes.

### PR DESCRIPTION
**Pull request by nsthorat**
Vector stores now return Iterators of np.array vectors.

### CODE REVIEW

File: lilac/embeddings/vector_store.py [hunk 1]
```diff
@@ -159,13 +159,10 @@ def get(self, keys: Iterable[PathKey]) -> Iterable[list[SpanVector]]:
       all_spans.append(spans)
       all_vector_keys.append([(*path_key, i) for i in range(len(spans))])
 
-    offset = 0
     flat_vector_keys = [key for vector_keys in all_vector_keys for key in (vector_keys or [])]
-    all_vectors = self._vector_store.get(flat_vector_keys)
+    all_vectors = iter(self._vector_store.get(flat_vector_keys))
```
Comment (brilee) [inline lilac/embeddings/vector_store.py:163]: don't think you need the iter?

File: lilac/embeddings/vector_store_hnsw.py [hunk 2]
```diff
@@ -105,15 +107,20 @@ def add(self, keys: list[VectorKey], embeddings: np.ndarray) -> None:
...
-        return np.array(self._index.get_items(self._key_to_label.values), dtype=np.float32)
-      locs = self._key_to_label.loc[cast(list[str], keys)].values
-      return np.array(self._index.get_items(locs), dtype=np.float32)
+        locs = self._key_to_label.values
+      else:
+        locs = self._key_to_label.loc[cast(list[str], keys)].values
+
+      for loc_chunk in chunks(locs, HNSW_RETRIEVAL_BATCH_SIZE):
+        chunk_items = np.array(self._index.get_items(loc_chunk), dtype=np.float32)
+        for vector in np.split(chunk_items, chunk_items.shape[0]):
```
Comment (brilee) [inline lilac/embeddings/vector_store_hnsw.py:122]: probably simpler/shorter:
```
for i in range(chunk_items.shape[0]):
  yield chunk_items[i]
```

**Review by brilee — APPROVED** [review]

File: lilac/embeddings/vector_store.py (hunk 1 above)
Comment (nsthorat) [reply lilac/embeddings/vector_store.py:163]: yep, no longer needed. was using islice but simplified it.

File: lilac/embeddings/vector_store_hnsw.py (hunk 2 above)
Comment (nsthorat) [reply lilac/embeddings/vector_store_hnsw.py:122]: need to use split here since it doesn't make a copy

**Review by dsmilkov — APPROVED** [review]

**Pull request merged**